lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -856,6 +856,109 @@ SwiftLanguage::GetHardcodedSynthetics() {
           }
           return nullptr;
         });
+    g_formatters.push_back([](lldb_private::ValueObject &valobj,
+                              lldb::DynamicValueType dyn_type,
+                              FormatManager &format_manager)
+                               -> lldb::SyntheticChildrenSP {
+      // Try to recognize a Swift type wrapped in a C++ interop wrapper class.
+      // These types have a typedef from a char to the swift mangled name, and a
+      // static constexpr char field whose type is the typedef, and whose name
+      // is $__swift_mangled_name.
+
+
+      // This only makes sense for Clang types.
+      auto type = valobj.GetCompilerType();
+      auto tsc = type.GetTypeSystem().dyn_cast_or_null<TypeSystemClang>();
+      if (!tsc)
+        return nullptr;
+
+      ProcessSP process_sp(valobj.GetProcessSP());
+      auto *swift_runtime = SwiftLanguageRuntime::Get(process_sp);
```

Comment:
This is the quickest check, whereas getting the full type might be some work.  Maybe do the swift runtime check first?